submissions/SergSenras/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,59 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+  Complete the below for code reviewers' convenience:
+
+  Code repository: _put repo URL here_
+  Web app: _put project's github pages URL here_
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Dweller {
+  constructor (species,name,gender,legs,saying,friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  getInfo(props) {
+    return props
+      .map( x => '<b>' + x + '</b>: ' + this[x] )
+      .join('; ');
+  }
+}
+
+class Animal extends Dweller {
+  constructor(species,name,gender,legs = 4,saying,friends) {
+    super(species,name,gender,legs,saying,friends);
+  }
+  
+  getInfo() {
+    return super.getInfo(['species','name','gender','legs','saying','friends']);
+  }
+}
+
+class Human extends Dweller {
+  constructor(name,gender,saying,friends,legs = 2,hands = 2) {
+    super('human',name,gender,legs,saying,friends);
+    this.hands = hands;
+  }
+
+  getInfo() {
+    return super.getInfo(['species','name','gender','hands','legs','saying','friends']);
```

Comment:
This way you **make** super class to know about hands.
Not OOP yet.
OOP compliant is to call `super.method()` and add hands.
No problem if hands come after anything else.
OOP compliance is even more important.

Reply:
So, if I understood you correctly base method now has fixed props and in `Human` class were added hands
```suggestion
  getInfo() {
    let humanProps = '; <b>hands:</b> ' + this.hands;
    return super.getInfo() + humanProps;
  }
```

Reply:
@SergSenras Perfectly correct.